src/web/tools/InteractionManager.ts

```diff
@@ -103,14 +102,15 @@ export default class InteractionManager {
   }
 
   public shouldHandlerBeCancelledBy(
-    _handler: GestureHandler,
-    otherHandler: GestureHandler
+    _handler: IGestureHandler,
+    otherHandler: IGestureHandler
   ): boolean {
-    return (
-      otherHandler instanceof NativeViewGestureHandler &&
-      otherHandler.getState() === State.ACTIVE &&
-      !otherHandler.isButton()
-    );
+    const isNativeHandler =
+      otherHandler.constructor.name === 'NativeViewGestureHandler';
+    const isActive = otherHandler.getState() === State.ACTIVE;
+    const isButton = !!otherHandler.isButton?.();
```

Comment:
```suggestion
    const isButton = otherHandler.isButton?.() === true;
```

Reply:
Done in [c392a94](https://github.com/software-mansion/react-native-gesture-handler/pull/2783/commits/c392a94089df0e392f4f910682b4154da4504fbb).